### PR TITLE
container handling implementation

### PR DESCRIPTION
**Pull request by rcohencyberarmor**
## Overview
container handler implementation
## Additional Information

> Any additional information that may be useful for reviewers to know 
-->

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -67,3 +67,27 @@ func (cfg *Config) GetFalcoKernelObjPath() string {
 func (cfg *Config) GetEbpfEngineLoaderPath() string {
 	return cfg.data.GetEbpfEngineLoaderPath()
 }
+
+func (cfg *Config) GetUpdateDataPeriod() int {
```
Comment (matthyx) [inline pkg/config/config.go:71]: I wonder if we should not directly return a duration here, I know you cannot unmarshall a duration, but it's not the client's responsibility to know he should convert int into seconds. WDYT?

File: pkg/config/config.go [hunk 2]
```diff
@@ -67,3 +67,27 @@ func (cfg *Config) GetFalcoKernelObjPath() string {
...
+	return cfg.data.GetUpdateDataPeriod()
+}
+
+func (cfg *Config) GetSniffingMaxTimes() int {
```
Comment (matthyx) [inline pkg/config/config.go:75]: same here

**Review by matthyx — COMMENTED** [review]
small nit regarding duration, otherwise let's move on and merge - this PR is too big

File: pkg/config/config.go (hunk 1 above)
Comment (rcohencyberarmor) [reply pkg/config/config.go:71]: so do you think it is better that we change the return here time.Second instead?

Comment (matthyx) [reply pkg/config/config.go:71]: return directly a time.duration that you can use for the ticker

File: pkg/conthandler/container_main_handler.go [hunk 3]
```diff
@@ -120,7 +120,7 @@ func (ch *ContainerHandler) startTimer(watchedContainer watchedContainerData, co
 }
 
 func createTicker() *time.Ticker {
-	return time.NewTicker(time.Duration(config.GetConfigurationConfigContext().GetUpdateDataPeriod()) * time.Second)
+	return time.NewTicker(time.Duration(config.GetConfigurationConfigContext().GetUpdateDataPeriod()))
```
Comment (matthyx) [inline pkg/conthandler/container_main_handler.go:123]: do you still need to convert to time.duration here?

Comment (rcohencyberarmor) [reply pkg/conthandler/container_main_handler.go]: forgot to push this file

File: pkg/conthandler/container_main_handler.go [hunk 4]
```diff
@@ -1,5 +1,214 @@
...
+	}
+	watchedContainer := containerDataInterface.(watchedContainerData)
+
+	err := watchedContainer.containerAggregator.StartAggregate(watchedContainer.syncChannel[StepEventAggregator])
+	if err != nil {
+		return
+	}
+
+	now := time.Now()
+	configStopTime := time.Duration(config.GetConfigurationConfigContext().GetSniffingMaxTimes())
```
Comment (matthyx) [inline pkg/conthandler/container_main_handler.go:140]: `time.Duration` conversion is not needed here

File: pkg/conthandler/v1/container_handler_data.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kubescape/k8s-interface/workloadinterface"
+)
+
+const (
+	CONTAINER_RUNNING = "Running"
```
Comment (matthyx) [inline pkg/conthandler/v1/container_handler_data.go:12]: snake case

File: pkg/storageclient/storage_client_mock.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+type StorageHttpClientMock struct {
+}
+
+const (
+	NGINX = "nginx"
+)
+
+var nginx_SBOM_spdx_bytes []byte
```
Comment (matthyx) [inline pkg/storageclient/storage_client_mock.go:17]: snake case

**Review by matthyx — APPROVED** [review]
let's merge

**Pull request merged**